lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1105,10 +1113,17 @@ void TNonblockingServer::listenSocket(THRIFT_SOCKET s) {
   serverSocket_ = s;
 
   if (!port_) {
-    sockaddr_in addr;
-    socklen_t size = sizeof(addr);
-    if (!getsockname(serverSocket_, reinterpret_cast<sockaddr*>(&addr), &size)) {
-      listenPort_ = ntohs(addr.sin_port);
+    #ifdef _WIN32
+      struct sockaddr_storage addr;
+      struct sockaddr_in *pAddr = (sockaddr_in *) &addr;
+      socklen_t size = sizeof(addr);
+    #else
+      struct sockaddr_in addr;
+      struct sockaddr_in *pAddr = &addr;
+      socklen_t size = sizeof(addr);
+    #endif
```

Comment:
OK, let's do it, then.
Here again, I prefer keeping your version only rather than `#ifdef` for the sake of readability over saving transitive ~100 bytes allocation, unless others disagree.
What do you think ? Do you have any concern ?